packages/dds/tree/src/feature-libraries/chunked-forest/codec/nodeShape.ts

```diff
@@ -27,17 +27,30 @@ import type { EncodedChunkShape, EncodedFieldShape, EncodedValueShape } from "./
 import { isStableId } from "@fluidframework/id-compressor/internal";
 
 export class NodeShape extends Shape<EncodedChunkShape> implements NodeEncoder {
-	// TODO: Ensure uniform chunks, encoding and identifier generation sort fields the same.
-	private readonly explicitKeys: Set<FieldKey>;
+	/**
+	 * Set of field keys that are encoded using specialized encoders.
+	 * TODO: Ensure uniform chunks, encoding and identifier generation sort fields the same.
+	 */
+	private readonly specializedFieldKeys: Set<FieldKey>;
 
 	public constructor(
 		public readonly type: undefined | TreeNodeSchemaIdentifier,
 		public readonly value: EncodedValueShape,
-		public readonly fieldEncoders: readonly KeyedFieldEncoder[],
-		public readonly extraLocal: undefined | FieldEncoder,
+		/**
+		 * Encoders for a specific set of fields, by key, in the order they will be encoded. These are fields for which
+		 * special encoding is required. For example, nested arrays.
```

Comment:
I don't think we have an official policy or style guide about it, but I prefer when writing markdown style documentation (like these doc comments) to put line breaks after sentences.
This makes line based editing tools (like diffs and suggestions) work better and less conflict prone.
Also "For example, nested arrays" isn't a sentence, and this isn't done because its required, its allowed because it can be an optimization. Its possible (and non schema aware encoder does it) to encode everything without these.

```suggestion
		 * Encoders for a specific set of fields, by key, in the order they will be encoded.
		 * These are fields for which specialized encoding is provided as an optimization.
		 * Using these for a given field instead of falling back to {@link NodeShape.specializedFieldEncoders} is often more efficient:
		 * this avoids the need to explicitly include the key and shape in the encoded data for each node instance.
		 * Instead this information is here, and thus is encoded only once as part of the node shape.
		 * These encoders will be used, even if the field they apply to is empty (which can add overhead for fields which are usually empty).
```

Reply:
Good points. Updated.